### PR TITLE
Updated outdated Dockerfile

### PR DESCRIPTION
**Pull request by eorituz**
Updated and slimed down the outdated Dockerfile

@andrevmatos Implemented the requested deletion of the .git folder

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-RUN pip install --upgrade pip
+RUN git clone https://github.com/raiden-network/raiden.git /app/raiden
 
-RUN git clone https://github.com/raiden-network/raiden.git /apps/raiden \
-    && cd /apps/raiden \
-    && pip install --upgrade -r requirements.txt \
-    && python setup.py develop
+RUN cd /app \
+    && pip3 install -c raiden/constraints.txt -e ./raiden \
+    && rm -fr raiden/.git
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply docker/Dockerfile]: It would actually be better to use the multi stage build:

https://docs.docker.com/develop/develop-images/multistage-build/

**Comment by eorituz** [comment]
@hackaugusto @andrevmatos Implemented the version as well as multi stage build.

**Comment by hackaugusto** [comment]
I'm good with the changes, maybe @andrevmatos wants to have a second look

**Comment by LefterisJP** [comment]
Lol everybody seems to approve nobody wants to merge. So I will go ahead and give this a push. :smile:

**Pull request merged**